nexus/go-handler.md

```diff
@@ -0,0 +1,307 @@
+# Temporal Go SDK - Nexus Handler
+
+## Overview
+
+An addition to the [Temporal Go SDK][temporal-go-sdk] that exposes the ability for users to handle Nexus API requests.
+
+**Goals**
+
+- Expose a simple, high level API for binding Nexus operations to common Temporal constructs
+- Support for starting and canceling operations
+- Support multi-tenant applications with minimal boilerplate
+
+**Non-Goals**
+
+- Support for getting an operation's result or info
+- API definitions for other SDKs
+- API for invoking Nexus calls - refer to the [Nexus Go SDK][nexus-go-sdk] for an out-of-workflow reference client
+- Server implementation details
+- Worker / server API contract
+- Allow advanced users to implement Nexus inbound and outbound request proxies for cross-organization calls where
+  data converters are not shared and data is not exposed to Temporal server.
+
+  This will quickly become a requirement for production use cases, especially with Temporal Cloud's zero trust model. It
+  will be addressed shortly after Nexus becomes available for preview.
+
+> NOTE: These are non-goals for _this_ proposal, but they will be defined in follow up proposals / PRs.
+
+## Background
+
+Nexus is a thin protocol on top of HTTP, defined in [this spec][nexus-http-api]. Temporal serves as an HTTP proxy,
+translating Nexus requests into tasks that can be polled on by an SDK worker on its configured task queue.
+Due to the synchronous nature of HTTP, the generated tasks are not persisted, and require an available worker
+to be processed.
+
+The process of translating requests into tasks and delivering them to workers is out of scope for this proposal.
```

Comment:
Yes, but it would help to clarify the properties that handler code need to respect.

For example, handlers need to be idempotent, if we want to trigger nexus operation from activities.
Handlers will also need to be deterministic if we want to do the triggering from a workflow, possibly by just storing the outcome, similar to an activity...
Also, if handlers are doing complex coordination we will quickly lose some of the nicest properties of temporal, it is best to use a single workflow to do the orchestration...

So all that means that handlers are going to be really simple, mostly a one-to-one mapping to workflow operations, and we should be clear about that upfront...

Reply:
Sure, I'll find a way to clarify some of that.